wpilibj/src/athena/java/edu/wpi/first/wpilibj/DifferentialDrive.java

```diff
@@ -0,0 +1,231 @@
+
+    y = limit(y);
+    y = applyDeadband(y, m_deadband);
+
+    rotation = limit(rotation);
+    rotation = applyDeadband(rotation, m_deadband);
+
+    double angularPower = 0.0;
+    double overPower;
+
+    if (isQuickTurn) {
+      if (Math.abs(y) < 0.2) {
+        final double alpha = 0.1;
+        m_quickStopAccumulator =
+            (1 - alpha) * m_quickStopAccumulator + alpha * limit(rotation) * 2;
+      }
+      overPower = 1.0;
+      angularPower = rotation;
+    } else {
+      overPower = 0.0;
+      angularPower = Math.abs(y) * rotation - m_quickStopAccumulator;
+
+      if (m_quickStopAccumulator > 1) {
+        m_quickStopAccumulator -= 1;
+      } else if (m_quickStopAccumulator < -1) {
+        m_quickStopAccumulator += 1;
+      } else {
+        m_quickStopAccumulator = 0.0;
+      }
+    }
+
+    double leftMotorOutput = y + angularPower;
+    double rightMotorOutput = y - angularPower;
+
+    // If overpowered rotation enabled
+    if (leftMotorOutput > 1.0) {
+      rightMotorOutput -= overPower * (leftMotorOutput - 1.0);
+      leftMotorOutput = 1.0;
+    } else if (rightMotorOutput > 1.0) {
+      rightMotorOutput = 1.0;
+      leftMotorOutput -= overPower * (rightMotorOutput - 1.0);
+    } else if (leftMotorOutput < -1.0) {
+      rightMotorOutput += overPower * (-1.0 - leftMotorOutput);
+      leftMotorOutput = -1.0;
+    } else if (rightMotorOutput < -1.0) {
+      leftMotorOutput += overPower * (-1.0 - rightMotorOutput);
+      rightMotorOutput = -1.0;
+    }
+
+    m_leftMotor.set(leftMotorOutput * m_maxOutput);
+    m_rightMotor.set(-rightMotorOutput * m_maxOutput);
+
+    m_safetyHelper.feed();
+  }
+
+  /**
+   * Tank drive method for differential-drive platform.
+   *
+   * @param left  The value to use for left side motors. [-1.0..1.0]
+   * @param right The value to use for right side motors. [-1.0..1.0]
+   */
+  public void tankDrive(double left, double right) {
+    tankDrive(left, right, true);
+  }
+
+  /**
+   * Tank drive method for differential-drive platform.
+   *
+   * @param left  The value to use for left side motors. [-1.0..1.0]
+   * @param right The value to use for right side motors. [-1.0..1.0]
+   * @param squaredInputs If set, increases the sensitivity at low speeds.
+   */
+  public void tankDrive(double left, double right, boolean squaredInputs) {
+    if (!m_reported) {
+      HAL.report(tResourceType.kResourceType_RobotDrive, 2, tInstances.kRobotDrive_Tank);
+      m_reported = true;
+    }
+
+    left = limit(left);
+    left = applyDeadband(left, m_deadband);
+
+    right = limit(right);
+    right = applyDeadband(right, m_deadband);
+
+    // square the inputs (while preserving the sign) to increase fine control
+    // while permitting full power
+    if (squaredInputs) {
+      left = Math.copySign(left * left, left);
+      right = Math.copySign(right * right, right);
+    }
+
+    m_leftMotor.set(limit(left) * m_maxOutput);
+    m_rightMotor.set(-limit(left) * m_maxOutput);
```

Comment:
1. Yep. I'll fix that.
2. Yes. For a robot that has a motor on each side wired with the same polarity, one will go the opposite direction. Assuming clockwise rotation as viewed from within the robot frame, the right side should be inverted to make the robot drive forward.